src/Majora/Framework/Loader/Bridge/Form/Type/EntityCollectionType.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+namespace Majora\Framework\Loader\Bridge\Form\Type;
+
+use Majora\Framework\Loader\Bridge\LoaderBridgeInterface;
+use Majora\Framework\Model\EntityCollection;
+use Symfony\Component\Form\AbstractType;
+use Symfony\Component\Form\Extension\Core\Type\ChoiceType;
+use Symfony\Component\OptionsResolver\Options;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+/**
+ * Class EntityCollectionType.
+ */
+class EntityCollectionType extends AbstractType
+{
+    /**
+     * @var array
+     */
+    private $loaders = [];
+
+    /**
+     * {@inheritdoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
+        parent::configureOptions($resolver);
+
+        $choices = function (Options $options) {
+            if ($this->hasLoader($options['loader_alias']) === false) {
+                throw new \RuntimeException(sprintf('Loader with alias "%s" not found.', $options['loader_alias']));
+            }
+
+            $loader = $this->getLoader($options['loader_alias']);
+
+            if (method_exists($loader, $options['loader_method']) === false) {
+                throw new \RuntimeException(
+                    sprintf('Method "%s" not found in "%s".', $options['loader_method'], get_class($loader))
+                );
+            };
+
+            $data = $loader->{$options['loader_method']}();
+
+            if (!($data instanceof EntityCollection)) {
+                throw new \RuntimeException(
+                    sprintf('The loader method should return an instance of "%s".', EntityCollection::class)
+                );
+            }
+
+            return $data;
+        };
+
+        $resolver
+            ->setRequired(
+                [
+                    'loader_alias',
+                    'loader_method',
+                ]
+            )
+            ->setAllowedTypes('loader_alias', 'string')
+            ->setAllowedTypes('loader_method', ['null', 'string'])
+            ->setDefaults(
+                [
+                    'choice_label' => function ($choice) {
+                        return (string)$choice;
+                    },
+                    'choice_value' => 'id',
+                    'choices' => $choices,
+                    'loader_method' => 'choiceList',
+                ]
+            );
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getParent()
+    {
+        return ChoiceType::class;
+    }
+
+    /**
+     * Register a loader.
+     *
+     * @param                       $alias
+     * @param LoaderBridgeInterface $loader
+     */
+    public function registerLoader($alias, LoaderBridgeInterface $loader)
+    {
+        if ($this->hasLoader($alias)) {
```

Comment:
The only limitation is that we can't have two loader with the same alias. The problem is if two loaders are added with the same alias we don't know which loader will override the other. 

May be we should add a priority (like for listeners) to be able to know the registration order and override in the good order. What do you think ?